File: lab4/lang-tests/test-3/assignment_forced_to_single.scala
object Assignment {

    def main (args: Array[String]) {
        
        val CONSTANT_m_assignment = 0;
        System.out.println("var multiple_assignment: "+CONSTANT_m_assignment);
        /* Dado que una constante no puede cambiar
        este programa no compilará */
        CONSTANT_m_assignment = 222;
    }
}

File: lab4/lang-tests/test-3/assignment_forced_to_single.py
# encoding: utf-8
class Constant(object):
    
    __UNPROTECTED = True
    value = None

    def __init__(self, value):
        super(Constant, self).__init__()
        self.value = value
        self.__UNPROTECTED = False

    def __setattr__(self, *arg):
        if not self.__UNPROTECTED:
            raise Exception("This value is only read")
        else:
            super(Constant, self).__setattr__(*arg)

a = Constant(2)
print a.value
# Dado que la clase Constant prohibe la reasignación de un valor
# Esto generará una excepción al cambiar el valor.
# Sin embargo, la variable a puede apuntar a otros valores :(
# de la forma a = "cambie de valor muajaja"
a.value = 3